src/session_handle.cpp

```diff
@@ -807,12 +812,14 @@ namespace libtorrent
 
 	proxy_settings session_handle::tracker_proxy() const
 	{
-		return proxy();
+		settings_pack const sett = get_settings();
+		return sett.get_bool(settings_pack::proxy_tracker_connections)
+			? proxy_settings(sett) : proxy();
```

Comment:
Do these not both do the same thing? Where instead it should return essentially proxy_settings(type=None)...